commit a7dfab7411cbf545f359dd3157e5df1eb0e7ce31	a7dfab7411cbf545f359dd3157e5df1eb0e7ce31
Author: Daniel Veillard <veillard@redhat.com>
Date:   Mon Feb 23 11:17:35 2015 +0800

    Stop parsing on entities boundaries errors
    
    For https://bugzilla.gnome.org/show_bug.cgi?id=744980
    
    There are times, like on unterminated entities that it's preferable to
    stop parsing, even if that means less error reporting. Entities are
    feeding the parser on further processing, and if they are ill defined
    then it's possible to get the parser to bug. Also do the same on
    Conditional Sections if the input is broken, as the structure of
    the document can't be guessed.

diff --git a/parser.c b/parser.c
index a8d1b673..bbe97eb5 100644
--- a/parser.c
+++ b/parser.c
@@ -5658,6 +5658,7 @@ xmlParseEntityDecl(xmlParserCtxtPtr ctxt) {
 	if (RAW != '>') {
 	    xmlFatalErrMsgStr(ctxt, XML_ERR_ENTITY_NOT_FINISHED,
 	            "xmlParseEntityDecl: entity %s not terminated\n", name);
+	    xmlStopParser(ctxt);
 	} else {
 	    if (input != ctxt->input) {
 		xmlFatalErrMsg(ctxt, XML_ERR_ENTITY_BOUNDARY,
